### ISSUE TITLE
Breaking sorting hopper does not drop a regular hopper

### ISSUE THREAD

**Issue by Drjoachim** [poiNt3D/SortingHopper2#4]
V 1.15.2 
Bukkit
if you break the sorting hopper it drops a hopper that has a different name, but still is a regular Hopper. The rename of the hopper should be executed on block break as well.

**Comment by poiNt3D** [comment]
I'm sorry, i cannot reproduce this bug. Can you check whether it's present if you don't have any other plugins?
As a temporary fix you can set
```
replacedrops: false
add_drop: COMPARATOR
```
to prevent players loosing resources.

**Comment by poiNt3D** [comment]
I get it now. You already have replacedrops set to false. So the plugin does nothing on block break. In previous versions of Minecraft hoppers always had the default name after breaking. Now they fixed it.
You should set replacedrops: true 
and it will drop modified hopper.

**Comment by poiNt3D** [comment]
I'm not sure i should fix that. Disable drop replacement feature was meant to make the plugin more compatible with various protection plugins and to avoid dupe. 
I better fix the hacky way it replaces drops (it wasn't possible at the time). This should make replacedrops safe to use.
```
//Hacky code
event.setCancelled(true);
event.getBlock().setType(Material.AIR);
event.getBlock().getWorld().dropItemNaturally(loc, drop);
```

**Comment by Drjoachim** [comment]
Not sure if that is the issue. 
Let me rephrase my previous one since it was not very detailed
> installed the plugin
> no changes on config
> create/give a player a sorter
> sorter is put in the world
> player breaks sorter and gets the following items:
>> comparator
>> a hopper with the name "item sorter"

The last is actually a regular hopper, so it is fine, yet it is a bit annoying that it does not stack with the regular hoppers as the name is different now.

I hope this is more clear. It is not really a breaking thing, just a bit an annoyance for my inventory

**Comment by poiNt3D** [comment]
Ok, the issue is clear. 
There is two ways you can handle drops with this plugin.
a) Replace drops: it cancels block break event completely, then drops custom item and manually replaces block with AIR.
b) Don't interfere. It will just make an additional drop. Original drop will be handled by Bukkit. It wasn't an issue before, since broken hoppers had lost their names after breaking.

The possible solutions are:
a) Make replacing drops less agressive - not sure if possible
b) Check compatibility with other plugins that cancel block breaking - lots of work
c) Make just renamed hoppers work as sorters

I will stick with 'c' for now.
As a server owner, if you not too worried about protection problems or you checked and there is none, you should just use replacedrops: true.

**Comment by poiNt3D** [comment]
New version will only check hopper name by default.
